### ISSUE TITLE
Related data fixes/enhancements

### ISSUE THREAD

**Issue by davmlaw** [SACGF/variantgrid#28]
* trios made from sub cohorts don’t seem to show
* Related X for sample - patients and analyses should have icons
* Classifications grid should show sample ID / name as option - turn on in related data (good for VCF with multiple samples)

Sarah RUNX1:

* untick increases number of tags (shows 4919 - all my skingsmith tags? vs 83 for all users). Patient page related data

**Comment by davmlaw** [comment]
🤖 Written by Claude

Went through the five items in this issue against current `master` - most have been overtaken by other work.

**Already done, nothing left to do**

* *Classifications grid should show sample ID / name as option* - done. `ClassificationColumns` has `sample_id` (hidden, used for links) and `sample_name` columns, gated on `settings.CLASSIFICATION_GRID_SHOW_SAMPLE` (default on, off for Shariant). The related data classification table uses that config, so it's already available there.
* *Trios made from sub cohorts don't seem to show* - fixed for the sample / VCF / patient pages: `related_data_for_samples` finds trios via `CohortSample` ids, which covers sub cohorts. Confirmed with a test. The parent **cohort** page was still missing them - see below.
* *Untick increases number of tags (4919 vs 83)* - not reproducible. Variant tags are now a DataTable scoped to the related analyses (`analysis_ids`) and permission filtered through `VariantTag.filter_for_user`, with an explicit "show group data" toggle. Treating this one as stale.
* *(comment) `related_data_for_samples` + `related_analyses_for_samples` duplicate a lot of queries* - the N+1 growth was fixed in #1590 / #1267 and locked in by `snpdb/tests/test_query_counts.py`. The two tags still each derive cohorts/trios/pedigrees, but it's a constant handful of queries now, so merging them isn't worth the churn.

**Fixed now**

* Cohort page related data listed only `cohort.trio_set` / `cohort.pedigree_set`, so a trio or pedigree built off a sub cohort was invisible on the parent cohort (and VCF) page, even though the sub cohorts themselves were listed. `related_data_for_cohort` and `related_analyses_for_cohort` now look at the cohort plus its sub cohorts.
* *Related X should have icons* - the preview system arrived after this issue was written and every model here defines `preview_icon`, so the related data lists now render it: cohorts, trios, pedigrees, samples, VCFs, specimens, cases and analyses.

**Comment by davmlaw** [comment]
VG testing - don't like icons, will use the SVG ones from analysis
